torchhd/datasets/collection_datasets.py

```diff
@@ -0,0 +1,403 @@
+import os
+import os.path
+from typing import Callable, Optional, Tuple, List
+import torch
+from torch.utils import data
+import pandas as pd
+import tarfile
+import numpy as np
+
+from .utils import download_file_from_google_drive
+
+
+class CollectionDataset(data.Dataset):
+    """`Generic class for loading datasets used in `Do we Need Hundreds of Classifiers to Solve Real World Classification Problems? <https://jmlr.org/papers/v15/delgado14a.html>`_.
+
+    Args:
+        dataset (string): Name of the dataset. Used when extracting the corresponding files from archive.
+        root (string): Root directory containing the files of the dataset.
+        has_test (bool): Indicates whether the dataset has a separate test file or not.
+        download (bool): If True, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    """
+
```

Comment:
You can set the `name: str` property to indicate that subclasses need to specify it. The `__dataset` is then not needed anymore.